nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/CheckBoxController.java

```diff
@@ -132,7 +140,7 @@ public void refresh(CheckBox checkbox, int index) {
         Set<Object> modelValues = retrieveModelValues();
         checkbox.setChecked(
                 modelValues.contains(
-                        areValuesDefined() ? checkbox.getText() : index
+                        !areValuesDefined() ? checkbox.getText() : values[index]
```

Comment:
What's the exact issue you're trying to solve? The current behavior is already the same as `SelectionController` as far as I can tell.

The current implementation is based on the documented behavior (see comments in the constructor). Specifically, if the `useItemsAsValues` param that is passed in to the constructor is `false`, the `CheckBoxController` will represent the selected fields using the 0-based index (as in integer). When you want to read the values in your UI, you can perform the index-to-value mapping if needed.

Let me know if I'm missing something.